documentation/docs/05-misc/02-testing.md

```diff
@@ -2,7 +2,156 @@
 title: Testing
 ---
 
-- component testing basics
-- rune testing basics
-- vitest setup
-- e2e
+Testing helps you write and maintain your code and guard against regressions. Testing frameworks help you with that, allowing you to describe assertions or expectations about how your code should behave. Svelte is unopinionated about which testing framework you use — you can write unit tests, integration tests, and end-to-end tests using solutions like Vitest, Jasmine, Cypress and Playwright.
+
+## Unit and integration testing using Vitest
+
+Unit tests allow you to test small isolated parts of your code. Integration tests allow you to test parts of your application to see if they work together. If you're using Vite (including via SvelteKit), we recommend using [Vitest](https://vitest.dev/).
+
+To get started, install Vitest and jsdom (a library that shims DOM APIs):
+
+```bash
+npm install -D vitest jsdom
+```
+
+Then adjust your `vite.config.js`:
+
+```js
+/// file: vite.config.js
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+	plugins: [
+		/* ... */
+	],
+	test: {
+		// If you are testing components client-side, you need to setup a DOM environment.
+		// If not all your files should have this environment, you can use a
+		// `// @vitest-environment jsdom` comment at the top of the test files instead.
+		environment: 'jsdom'
+	},
+	// Tell Vitest to prefer the browser condition. Svelte uses a so-called exports map
+	// which loads different code depending on the environment - we want to load the browser environment.
+	resolve: process.env.VITEST
+		? {
+				conditions: ['browser']
+			}
+		: undefined
+});
+```
+
+After that, you can create a test file in which you import the component to test, then interact with it programmatically and write expectations about the results:
+
+```js
+/// file: component.test.js
```

Comment:
if we're using non-null assertions (`querySelector(...)!.click()`) then this needs to be a `.ts` file 

```suggestion
/// file: component.test.ts
```